src/player.cpp

```diff
@@ -10884,42 +10884,45 @@ bool player::has_enough_charges( const item &it, bool show_msg ) const
     return true;
 }
 
-bool player::consume_charges(item *used, long charges_used)
+bool player::consume_charges( item& used, long qty )
 {
-    // Non-tools can use charges too - when they're comestibles
-    const auto tool = dynamic_cast<const it_tool*>(used->type);
-    const auto comest = dynamic_cast<const it_comest*>(used->type);
-    if( charges_used <= 0 || (tool == nullptr && comest == nullptr) ) {
```

Comment:
There is another funamental change in the behavior of the function here: if `charges_used` is 0 (or negative), the function would just return without doing anything.

This is used for example when applying a crowbar. The isue function (called from `invoke_item`) returns `itype::charges_to_use`, which is 0 for crowbars (see JSON of it). That value is forward to this function here and would previously not do anything. Now it does remove the crowbar.